contrib/sbtc/token-key-admin/contracts/sbtc-token-key-admin.clar

```diff
@@ -0,0 +1,186 @@
+;; summary:
+;; description:
+
+;; traits
+;;
+(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)
+
+;; token definitions
+;; 
+(define-fungible-token sbtc u21000000)
+
+;; constants
+;;
+(define-constant err-invalid-caller u1)
+(define-constant err-invalid-signer-id u2)
+(define-constant err-not-token-owner u3)
+
+;; data vars
+;;
+(define-data-var contract-owner principal tx-sender)
+(define-data-var coordinator (optional {addr: principal, key: (buff 33)}) none)
+(define-data-var num-keys uint u4000)
+(define-data-var num-parties uint u4000)
+(define-data-var threshold uint u2800)
+(define-data-var bitcoin-payout-address (optional (string-ascii 72)) none)
+
+;; data maps
+;;
+(define-map signers uint {addr: principal, key: (buff 33)})
+
+;; public functions
+;;
+(define-public (set-contract-owner (owner principal))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set contract-owner owner))
+    )
+)
+
+(define-public (set-coordinator-data (data {addr: principal, key: (buff 33)}))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set coordinator (some data)))
+    )
+)
+
+(define-public (set-num-keys (n uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set num-keys n))
+    )
+)
+
+(define-public (set-num-parties (n uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set num-parties n))
+    )
+)
+
+(define-public (set-threshold (n uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set threshold n))
+    )
+)
+
+(define-public (set-signer-data (id uint) (data {addr: principal, key: (buff 33)}))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (asserts! (is-valid-signer-id id) (err err-invalid-signer-id))
+        (ok (map-set signers id data))
+    )
+)
+
+(define-public (delete-signer-data (id uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (asserts! (is-valid-signer-id id) (err err-invalid-signer-id))
+        (ok (map-delete signers id))
+    )
+)
+
+(define-public (set-bitcoin-payout-address (addr (string-ascii 72)))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set bitcoin-payout-address (some addr)))
+    )
+)
+
+(define-public (mint! (amount uint))
+    (begin
+        (asserts! (is-coordinator) (err err-invalid-caller))
+        (ft-mint? sbtc amount tx-sender)
+    )
+)
+
+(define-public (burn! (amount uint))
+    (begin
+        (asserts! (is-coordinator) (err err-invalid-caller))
+        (ft-burn? sbtc amount tx-sender)
+    )
+)
+
+(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+	(begin
+		(asserts! (is-eq tx-sender sender) (err err-not-token-owner))
```

Comment:
At some point, we would like to let users delegate transfer rights to contracts. I'm certain we'll get this feedback when testnet goes live but we can address in the next contract iteration.

Reply:
How would this work exactly?  transfer the FTs to a contract principal?  Or something internal like a delegate map?

Def not needed for this PR in any case.

Reply:
The latter. I believe this could be addressed with a new SIP trait: https://github.com/stacksgov/sips/pull/84